### PR TITLE
Chapter1/homework/3

### PR DESCRIPTION
**Pull request by HellHunterMax**

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4864051&bid=16132282)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            Console.Write(message);
+            return int.Parse(Console.ReadLine());
+        }
+        public static float FloatPrompter(string message)
+        {
+            Console.Write(message);
+            return float.Parse(Console.ReadLine());
+        }
+        public static float BMICalculator(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'BMICalculator' to match pascal case naming rules, consider using 'BmiCalculator'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4864051)

**Comment by HellHunterMax** [comment]
> Solid solution in my eyes. You used a lot of nouns for your function names, though.
> Nouns serve a different purpose - are reserved for a different meaning by convention - so please change those names to be verbs instead. :)

I do not understand the difference between  PromtFloat or FloatPrompter? aren't they both a verb ?
 I do get why it should be the other way around though for readability. And that's why i changed it.

Or is this not what you meant ? Thanks for the review !

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4864051&bid=16132282)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            string firstName = PromptString("What is your first name: ");
+            string lastName = PromptString("What is your last name: ");
+            int age = PromptInt("What is your age: ");
+            float weight = PromptFloat("what is your weight in Kg: ");
+            float height = PromptFloat("what is your height in cm: ");
+            Console.WriteLine($"{firstName} {lastName} is {age} years old, his weight is {weight} kg and his height is {height} cm. ");
+            float bmi = BmiCalculator(weight, height / 100);
+            Console.WriteLine($"Your BMI is {bmi}");
+        }
```
Comment (diverges) [inline Src/BootCamp.Chapter/Lesson3.cs]: Add a space between your function declarations to improve readability.

Comment (HellHunterMax) [reply Src/BootCamp.Chapter/Lesson3.cs]: ah yes i see and agree! thanks for the tip.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4864051&bid=16132282)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
+        {
+            PersonInfoAskerAndPrinter();
+            PersonInfoAskerAndPrinter();
+        }
+        private static void PersonInfoAskerAndPrinter()
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: This also needs a change.
To explain it for the last comment and for this one:

If you would imagine something when reading "Asker" or "Printer" you would think of some object or being that performs this task. A **printer** IRL **prints** documents that you want to be printed. Note the difference!

A good name could be "PromptPersonData()" or "ProcessPerson()" or something like that. Feel free to ask again if you need more details on that subject. :)

Comment (HellHunterMax) [reply Src/BootCamp.Chapter/Lesson3.cs]: Oh thank you so much i missed that one.
thanks for explaining as wel very helpful.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4864051&bid=16132282)

**Pull request closed without merging**